sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -219,13 +219,21 @@ object JdbcUtils extends Logging {
     val sb = new StringBuilder()
     val dialect = JdbcDialects.get(url)
     df.schema.fields foreach { field => {
-      val name = field.name
+      val name = dialect.quoteIdentifier(field.name)
       val typ: String = getJdbcType(field.dataType, dialect).databaseTypeDefinition
       val nullable = if (field.nullable) "" else "NOT NULL"
       sb.append(s", $name $typ $nullable")
     }}
     if (sb.length < 2) "" else sb.substring(2)
   }
+  
+  /**
+   * Compute the table name string for this RDD.
+   */
+  def tableString(table: String, url: String): String = {
```

Comment:
@JoshRosen Are you suggesting to build similar logic as spark-redshift has some fancy logic to handle all of these corner-cases; that you highlighted databricks/spark-redshift#102 as to how the TableName wrapper works.

Please let me know ur thoughts